### ISSUE TITLE
Completely distorted GUI on Windows 10

### ISSUE THREAD

**Issue by mark-orion** [mtytel/helm#252]
I am running Windows 10 as a Virtualbox guest. However all other programs work fine, only Helm offers this kind of very distorted GUI:

![Screenshot from 2020-06-12 18-26-40](https://user-images.githubusercontent.com/5022007/84530045-6379dc00-acda-11ea-8f3c-7f6f84ee1eb8.png)

**Comment by brunobeltran** [comment]
I can confirm that this is also the case for me on a native install of Windows 10, update 1909.

**Comment by DavidsonAudioMulti** [comment]
This has to do with DPI awareness if you change the windows Display preferences Scale and Layout to 100% it resolves this issue. I wonder what the best way to fix this is??

**Comment by jtommi** [comment]
@DavidsonAudioMulti I can confirm that setting Scale to 100% fixes the UI. Unfortunately not viable for me.
I might get a second monitor that would have scaling disabled.
For those that don't know where to change that : [View display settings in Windows 10](https://support.microsoft.com/en-us/windows/view-display-settings-in-windows-10-37f0e05e-98a9-474c-317a-e85422daa8bb)

**Comment by HelgenX** [comment]
Yeah, you change your scaling to 100%, unfortunately that's not a fix to this, especially for people who can't see tiny text on the screen, and it seems the developers will never fix it.

**Comment by zmweske** [comment]
I had this problem on windows 10 but it seems to be working at 110% scaling on windows 11 for me, just a heads up to anyone else interested

**Comment by CRTX** [comment]
Confirmed this is a scaling issue. Having three 4k screens sucks when it comes to using this plugin lol. Setting scaling back to 100% works

**Comment by karl-police** [comment]
I have the same issue when the scaling setting is something else than 100% I guess. Only when used as a plugin, this doesn't happen if you open the application directly.

**Comment by spdrcd** [comment]
> I had this problem on windows 10 but it seems to be working at 110% scaling on windows 11 for me, just a heads up to anyone else interested

I might upgrade to windows 11 possibly because of this...

**Comment by spdrcd** [comment]
> I have the same issue when the scaling setting is something else than 100% I guess. Only when used as a plugin, this doesn't happen if you open the application directly.

can confirm this as well :)... it happend to me when i open this in bitwig studio

**Comment by twobob** [comment]
I havent really looked. nor is this my project.
but  I noted in the source code the concept of a "constrainer" - perhaps there is already a low cost way of forcing the smaller layout.

TBH I use win 11 and I use large fonts so I just can't test this but seems doable

**Comment by spdrcd** [comment]
> I havent really looked. nor is this my project. but I noted in the source code the concept of a "constrainer" - perhaps there is already a low cost way of forcing the smaller layout.
> 
> TBH I use win 11 and I use large fonts so I just can't test this but seems doable

let me know what part of the code where that is! maybe i can go in and make edits

**Comment by twobob** [comment]
> > I havent really looked. nor is this my project. but I noted in the source code the concept of a "constrainer" - perhaps there is already a low cost way of forcing the smaller layout.
> > TBH I use win 11 and I use large fonts so I just can't test this but seems doable
> 
> let me know what part of the code where that is! maybe i can go in and make edits

https://github.com/search?q=repo%3Amtytel%2Fhelm%20constrainer&type=code
and maybe https://github.com/mtytel/helm/commit/5c2a57dc6a79a23a68b924303882468ddfa29ac5 to get a sense of one of the later (last) constrainer implementations

**Comment by spdrcd** [comment]
> I havent really looked. nor is this my project. but I noted in the source code the concept of a "constrainer" - perhaps there is already a low cost way of forcing the smaller layout.`
> 
> TBH I use win 11 and I use large fonts so I just can't test this but seems doable

Ok it seems like i can reproduce the issue even after upgrading to windows 11... happens when its within a DAW as a hosted VST with windows scaling more than `100%` for mine device its set to `150%` by default... 
Standalone still works. It does also seem like PRs are very slow to merge kinda at a limbo state... 

@mtytel 's `vital` synth does look legit and doesnt have scaling issues, i might just use that
